### PR TITLE
Tests - Second take 

### PR DESCRIPTION
**Pull request by osdnk**

### CODE REVIEW

File: Example/test/e2e/sample.spec.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+describe('Example', () => {
+  beforeEach(async () => {
+    if (typeof device === 'undefined') {
```
Comment (mironiasty) [inline Example/test/e2e/sample.spec.js]: 😱

Comment (osdnk) [reply Example/test/e2e/sample.spec.js]: 😱

File: Example/test/e2e/init.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const detox = require('detox');
+const config = require('../../package.json').detox;
+const adapter = require('detox/runners/jest/adapter');
+jest.setTimeout(120000);
+jasmine.getEnv().addReporter(adapter);
+beforeAll(async () => {
+  await detox.init(config);
+});
+beforeEach(async () => {
```
[the remark this answers is not in the document]
Comment (osdnk) [reply Example/test/e2e/init.js]: Not sure! Gonna check it whether rest of changes will give positive result.

File: Example/test/e2e/init.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const detox = require('detox');
+const config = require('../../package.json').detox;
+const adapter = require('detox/runners/jest/adapter');
+jest.setTimeout(120000);
+jasmine.getEnv().addReporter(adapter);
+beforeAll(async () => {
```
[the remark this answers is not in the document]
Comment (osdnk) [reply Example/test/e2e/init.js]: Nope, it's `jest`, not `mocha`

File: Example/test/e2e/init.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  await detox.init(config);
+});
+beforeEach(async () => {
+  await adapter.beforeEach();
+});
+afterAll(async () => {
```
[the remark this answers is not in the document]
Comment (osdnk) [reply Example/test/e2e/init.js]: same ^

**Pull request closed without merging**